packages/transmute-framework/contracts/TransmuteFramework/EventStoreFactory.sol

```diff
@@ -0,0 +1,55 @@
+pragma solidity ^0.4.11;
+
+import "./EventStore.sol";
+import "./SetLib/AddressSet/AddressSetLib.sol";
+
+contract EventStoreFactory is EventStore {
+
+  using AddressSetLib for AddressSetLib.AddressSet;
+
+  mapping (address => AddressSetLib.AddressSet) creatorEventStoreMapping;
+  AddressSetLib.AddressSet EventStoreAddresses;
+
+  // Modifiers
+  modifier eventStoreExists(address _eventStoreAddress) {
+    require(EventStoreAddresses.contains(_eventStoreAddress));
+    _;
+  }
+
+  // Fallback Function
+  function () public payable { revert(); }
+
+  // Constructor
+  function EventStoreFactory() public payable { }
+
+  // Interface
+  function createEventStore(address[] _whitelist) public returns (address) {
+    EventStore newEventStore = new EventStore();
+
+    newEventStore.setWhitelist(_whitelist);
```

Comment:
technically, this is not in the ES contructor, why not pass the whitelist to the constructor?

Reply:
If you do that truffle says you don't have enough gas... in fact, I tried passing anything to the constructor and got the same error. I tried figuring out what was going on for hours and settled with this because it works. 

What you suggest was and still is the intended implementation, though.